Refresh.GameServer/Endpoints/Game/Playlists/PlaylistEndpoints.cs

```diff
@@ -0,0 +1,338 @@
+        //                                             accidentally falling outside of the branch during a possible future
+        //                                             refactor and returning OK when not intended. ok, rider? 
+        else
+        {
+            GameLevel? level = database.GetLevelByIdAndType(slotType, slotId);
+
+            // If the level doesn't exist, exit gracefully
+            if (level == null)
+                return NotFound;
+
+            // Add the level to the playlist    
+            database.AddLevelToPlaylist(level, parentPlaylist);
+            
+            return OK;
+        }
+    }
+
+    [GameEndpoint("removeFromPlaylist/{playlistId}", HttpMethods.Post)]
+    [RequireEmailVerified]
+    public Response RemoveSlotFromPlaylist(RequestContext context, GameDatabaseContext database, GameUser user,
+        int playlistId)
+    {
+        // Extract the slot type, ensuring its set
+        string? slotType = context.QueryString["slot_type"];
+        if (slotType == null)
+            return BadRequest;
+        
+        // Extract the slot ID, ensuring its valid
+        if (!int.TryParse(context.QueryString["slot_id"], out int slotId))
+            return BadRequest;
+
+        GamePlaylist? parentPlaylist = database.GetPlaylistById(playlistId);
+
+        // If the parent doesn't exist, exit gracefully
+        if (parentPlaylist == null)
+            return NotFound;
+
+        // Dont let people remove slots from other's playlists
+        if (parentPlaylist.Creator.UserId != user.UserId)
+            return Unauthorized;
+
+        // Special handling for playlists
+        if (slotType == "playlist")
+        {
+            GamePlaylist? childPlaylist = database.GetPlaylistById(slotId);
+
+            // If the child doesn't exist, exit gracefully
+            if (childPlaylist == null)
+                return NotFound;
+            
+            // Remove the playlist from the parent
+            database.RemovePlaylistFromPlaylist(childPlaylist, parentPlaylist);
+
+            // ReSharper disable once ExtractCommonBranchingCode see like 3 lines below (line count subject to change)
+            return OK;
+        }
+        // ReSharper disable once RedundantIfElseBlock i am intentionally writing this code like this to prevent code
+        //                                             accidentally falling outside of the branch during a possible future
+        //                                             refactor and returning OK when not intended. ok, rider? 
+        else
+        {
+            GameLevel? level = database.GetLevelByIdAndType(slotType, slotId);
+
+            // If the level doesn't exist, exit gracefully
+            if (level == null)
+                return NotFound;
+
+            // Remove the level from the playlist
+            database.RemoveLevelFromPlaylist(level, parentPlaylist);
+            
+            return OK;
+        }
```

Comment:
```suggestion
        string? slotType = context.QueryString["slot_type"];
        if (slotType == null)
            return BadRequest;
        
        if (!int.TryParse(context.QueryString["slot_id"], out int slotId))
            return BadRequest;

        GamePlaylist? parentPlaylist = database.GetPlaylistById(playlistId);
        if (parentPlaylist == null)
            return NotFound;

        // Dont let people remove slots from other's playlists
        if (parentPlaylist.Creator.UserId != user.UserId)
            return Unauthorized;

        // Removing a playlist from a playlist requires a special case, since we use `SubPlaylistRelation` internally to record child playlists.
        if (slotType == "playlist")
        {
            GamePlaylist? childPlaylist = database.GetPlaylistById(slotId);
            if (childPlaylist == null)
                return NotFound;
            

            database.RemovePlaylistFromPlaylist(childPlaylist, parentPlaylist);

            // ReSharper disable once ExtractCommonBranchingCode see like 3 lines below (line count subject to change)
            return OK;
        }
        // ReSharper disable once RedundantIfElseBlock i am intentionally writing this code like this to prevent code
        //                                             accidentally falling outside of the branch during a possible future
        //                                             refactor and returning OK when not intended. ok, rider? 
        else
        {
            GameLevel? level = database.GetLevelByIdAndType(slotType, slotId);
            if (level == null)
                return NotFound;

            database.RemoveLevelFromPlaylist(level, parentPlaylist);
            
            return OK;
        }
```